core/src/main/java/org/apache/carbondata/core/scan/filter/executer/RowLevelRangeLessThanEqualFilterExecuterImpl.java

```diff
@@ -262,10 +259,7 @@ private BitSet getFilteredIndexes(DimensionColumnDataChunk dimensionColumnDataCh
       int numerOfRows) {
     byte[] defaultValue = null;
     if (dimColEvaluatorInfoList.get(0).getDimension().hasEncoding(Encoding.DIRECT_DICTIONARY)) {
-      DirectDictionaryGenerator directDictionaryGenerator = DirectDictionaryKeyGeneratorFactory
-          .getDirectDictionaryGenerator(
-              dimColEvaluatorInfoList.get(0).getDimension().getDataType());
-      int key = directDictionaryGenerator.generateDirectSurrogateKey(null) + 1;
+      int key = 0;
```

Comment:
I think it is better to get null key from respective interfaces instead of hard code. if the null value key changes in future also no effect here

Reply:
And also the null value for dictionary is 1 not 0